sql/core/src/test/scala/org/apache/spark/sql/CollationSQLExpressionsSuite.scala

```diff
@@ -2434,829 +2447,622 @@ class CollationSQLExpressionsSuite
     )
   }
 
-  test("min_by supports collation") {
-    val collation = "UNICODE"
-    val query = s"SELECT min_by(x, y) FROM VALUES ('a', 10), ('b', 50), ('c', 20) AS tab(x, y);"
+  // common method for subsequent tests verifying various SQL expressions with collations
+  private def testCollationSqlExpressionCommon(
+      query: String,
+      collation: String,
+      result: Seq[Row],
+      dataType: DataType): Unit = {
     withSQLConf(SqlApiConf.DEFAULT_COLLATION -> collation) {
-      checkAnswer(
-        sql(query),
-        Seq(
-          Row("a")
-        )
-      )
+      // check result correctness
+      checkAnswer(sql(query), result)
       // check result row data type
-      val dataType = StringType(collation)
       assert(sql(query).schema.head.dataType == dataType)
     }
   }
 
+  test("min_by supports collation") {
+    testAdditionalCollations.foreach { collation =>
+      val query = "SELECT min_by(x, y) FROM VALUES ('a', 10), ('b', 50), ('c', 20) AS tab(x, y);"
+      val result = Seq(
+        Row("a")
+      )
+      val dataType = StringType(collation)
+      testCollationSqlExpressionCommon(query, collation, result, dataType)
+    }
+  }
```

Comment:
```suggestion
  test("min_by supports collation") {
    testAdditionalCollations.foreach { collation =>
      testCollationSqlExpressionCommon(
        query = "SELECT min_by(x, y) FROM VALUES ('a', 10), ('b', 50), ('c', 20) AS tab(x, y);",
        collation,
        dataType = StringType(collation),
        result = Seq(Row("a")))
    }
  }
```

Reply:
Makes sense, applied the suggestion, thanks!